web/src/providers/toast.tsx

```diff
@@ -0,0 +1,84 @@
+"use client";
+
+import React, { createContext, useContext, useState } from "react";
+import * as Toast from "@radix-ui/react-toast";
+import {
+	InfoCircledIcon,
+	ExclamationTriangleIcon,
+	CheckCircledIcon,
+} from "@radix-ui/react-icons";
+import { Flex } from "@radix-ui/themes";
+
+type ToastType = "success" | "error" | "info";
```

Comment:
Could be an enum:

```suggestion
enum ToastType {
    SUCCESS = "Success",
    ERROR = "Error",
    INFO = "Info",
}
```

By already assigning it to a string value with an uppercase first letter, you then don't need to do `{type.charAt(0).toUpperCase() + type.slice(1)}`, you can just use do `{type}` directly.